### PR TITLE
Fix other entities tests

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java [hunk 1]
```diff
@@ -208,10 +208,11 @@ public Experiment insertExperiment(Experiment experiment, UserInfo userInfo)
     createRoleBindingsForExperiment(experiment, userInfo);
     try (Session session = modelDBHibernateUtil.getSessionFactory().openSession()) {
       Transaction transaction = session.beginTransaction();
-      session.save(RdbmsUtils.generateExperimentEntity(experiment));
+      ExperimentEntity experimentEntity = RdbmsUtils.generateExperimentEntity(experiment);
+      session.save(experimentEntity);
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java:212]: return experiment without sorting fields so need to call toProtos for sorting fields

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 2]
```diff
@@ -510,7 +512,7 @@ private RepositoryEntity setRepository(
       }
     }
     session.getTransaction().commit();
-    if (create) {
+    if (create || nameChanged) {
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:515]: While updating repository name we should have to update name at UAC as well which is currently not happened so now fixed it with calling `setResource` if request found for update name

File: backend/src/test/java/ai/verta/modeldb/DatasetTest.java [hunk 3]
```diff
@@ -1414,6 +1413,9 @@ public void getLastExperimentByDataset() throws InterruptedException {
           experimentRun.getDateUpdated(),
           response.getExperimentRun().getDateUpdated());
 
+      ReconcilerInitializer.initialize(
+          testConfig, services, testConfig.getTestJdbi(), handleExecutor);
+
```
Comment (anandjakhaniya) [inline backend/src/test/java/ai/verta/modeldb/DatasetTest.java:1418]: `LastExperimentByDatasetId` endpoint called based on update timestamp. Before changes we have inline update parent timestamp code  but now we have reconcile utils so need to run it before calling `LastExperimentByDatasetId`

File: backend/src/test/java/ai/verta/modeldb/DatasetTest.java [hunk 4]
```diff
@@ -1696,19 +1698,6 @@ public void createDatasetWithGlobalSharingOrganization() {
...
-            .setName(orgRoleName)
-            .setScope(RoleScope.newBuilder().setOrgId(organization.getId()).build())
-            .build();
-    GetRoleByName.Response getRoleByNameResponse =
-        roleServiceBlockingStub.getRoleByName(getRoleByName);
-    assertEquals(
-        "Expected role name not found in DB",
-        orgRoleName,
-        getRoleByNameResponse.getRole().getName());
-
```
Comment (anandjakhaniya) [inline backend/src/test/java/ai/verta/modeldb/DatasetTest.java:1711]: No need to validate internal rolebinding at MDB

File: backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java [hunk 5]
```diff
@@ -140,18 +138,6 @@ public void createProjectWithGlobalSharingOrganization() {
...
-        GetRoleByName.newBuilder()
-            .setName(orgRoleName)
-            .setScope(RoleScope.newBuilder().setOrgId(organization.getId()).build())
-            .build();
-    GetRoleByName.Response getRoleByNameResponse =
-        roleServiceBlockingStub.getRoleByName(getRoleByName);
-    assertEquals(
-        "Expected role name not found in DB",
-        orgRoleName,
-        getRoleByNameResponse.getRole().getName());
```
Comment (anandjakhaniya) [inline backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java:154]: No need to validate internal rolebinding at MDB

**Review by conradoverta — APPROVED** [review]

**Pull request merged**